lib/lf/uscalfe/loc_comp_ellbvp.h

```diff
@@ -346,8 +368,37 @@ class MassEdgeMatrixProvider {
         edge_sel_(std::move(edge_selector)),
         fe_precomp_() {
     auto fe = fe_space->ShapeFunctionLayout(base::RefEl::kSegment());
+    LF_ASSERT_MSG(fe != nullptr, "No shape functions specified for edges");
+    // Precompute entity-independent quantities based on a LehrFEM++ built-in
+    // quadrature rule
     fe_precomp_ = PrecomputedScalarReferenceFiniteElement(
         fe, quad::make_QuadRule(base::RefEl::kSegment(), 2 * fe->Degree()));
+    LF_ASSERT_MSG(fe_precomp_.isInitialized(),"PFE object not initialized!");
+  }
+  /**
+   * @brief Constructor performing cell-independent initializations
+   *
+   * @param fe_space Describes the shapefunctions
+   * @param gamma coefficient function through functor object
+   * @param quadrule quadrature rule for EDGE entities
+   * @param edge_selector predicate object selecting active to be covered in
+   * the assembly
+   *
+   * This constructor takes a user-supplied quadrature rule.
+   */
+  MassEdgeMatrixProvider(std::shared_ptr<UniformScalarFESpace<SCALAR>> fe_space,
+                         COEFF gamma, lf::quad::QuadRule quadrule,
+                         EDGESELECTOR edge_selector = base::PredicateTrue{})
+      : gamma_(std::move(gamma)),
+        edge_sel_(std::move(edge_selector)),
+        fe_precomp_() {
+    auto fe = fe_space->ShapeFunctionLayout(base::RefEl::kSegment());
+    LF_ASSERT_MSG(fe != nullptr, "No shape functions specified for edges");
+    LF_ASSERT_MSG(quadrule.RefEl() == base::RefEl::kSegment(),
+                  "Quadrature rule not meant for EDGE entities!");
+    // Precompute entity-independent quantities
+    fe_precomp_ = PrecomputedScalarReferenceFiniteElement(fe, quadrule);
+    LF_ASSERT_MSG(fe_precomp_.isInitialized(),"PFE object not initialized!");
```

Comment:
I think this line is not needed since we check that `fe != nullptr` above and since fe_precomp_ is exlicitly initialized.